src/test/optimizer/strategy/predicate_pushdown_rule_test.cpp

```diff
@@ -138,4 +139,98 @@ TEST_F(PredicatePushdownRuleTest, ComplexBlockingPredicatesPushdownTest) {
   EXPECT_EQ(reordered->left_input()->right_input()->left_input(), _table_a);
 }
 
+TEST_F(PredicatePushdownRuleTest, AllowedValuePredicatePushdownThroughProjectionTest) {
+  auto int_float_node_a = StoredTableNode::make("a");
+  auto a = LQPColumnReference{int_float_node_a, ColumnID{0}};
+
+  auto parameter_c = value_(5);
+  auto lqp_c = AggregateNode::make(expression_vector(), expression_vector(max_(add_(a, parameter_c))),
+                                   ProjectionNode::make(expression_vector(add_(a, parameter_c)), int_float_node_a));
+
+  auto select_c = select_(lqp_c, std::make_pair(ParameterID{0}, a));
+
+  auto projection_node = ProjectionNode::make(expression_vector(_a_a, _a_b, select_c), _table_a);
+
+  auto predicate_node = std::make_shared<PredicateNode>(greater_than_(_a_a, _a_b));
+  predicate_node->set_left_input(projection_node);
+
+  auto reordered = StrategyBaseTest::apply_rule(_rule, predicate_node);
+
+  EXPECT_EQ(reordered, projection_node);
+  EXPECT_EQ(reordered->left_input(), predicate_node);
+  EXPECT_EQ(reordered->left_input()->left_input(), _table_a);
+}
+
+TEST_F(PredicatePushdownRuleTest, AllowedColumnPredicatePushdownThroughProjectionTest) {
+  auto int_float_node_a = StoredTableNode::make("a");
+  auto a = LQPColumnReference{int_float_node_a, ColumnID{0}};
+
+  auto parameter_c = parameter_(ParameterID{0}, a);
+  auto lqp_c = AggregateNode::make(expression_vector(), expression_vector(max_(add_(a, parameter_c))),
+                                   ProjectionNode::make(expression_vector(add_(a, parameter_c)), int_float_node_a));
+
+  auto select_c = select_(lqp_c, std::make_pair(ParameterID{0}, a));
+
+  auto projection_node = ProjectionNode::make(expression_vector(_a_a, _a_b, select_c), _table_a);
+
+  auto predicate_node = std::make_shared<PredicateNode>(greater_than_(_a_a, _a_b));
+  predicate_node->set_left_input(projection_node);
+
+  auto reordered = StrategyBaseTest::apply_rule(_rule, predicate_node);
+
+  EXPECT_EQ(reordered, projection_node);
+  EXPECT_EQ(reordered->left_input(), predicate_node);
+  EXPECT_EQ(reordered->left_input()->left_input(), _table_a);
+}
+
+TEST_F(PredicatePushdownRuleTest, ForbiddenPredicatePushdownThroughProjectionTest) {
```

Comment:
Not completely sure if I get this example right -- I cannot push the Predicate under the Projection because the Projection is the one that makes the TableScan (that the Predicate refers to) available to the Predicate?

And it could theoretically be made possible if we had more logic that allowed us to e.g. split the Projection into two Projections, one making `select_c` and `_a_b` available to the Predicate and one doing e.g. super expensive subqueries to compute `_a_a`?